### PR TITLE
Fix multiple issues with RadarUpdater from the quick info window

### LINKED ISSUE #5533: Hero's dialog dissappers after right clicking on a hero
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

Open a castle with a hero. Open the hero's window. Close the window. Open the window again. Right click on a hero's portrait to see a popup window and this what happens:
image

### Save file

Any save will do.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #5533

Also this PR fixes a ton of issues with `RadarUpdater` in the "no interface" mode. Currently in the `master` branch:

https://user-images.githubusercontent.com/32623900/173959820-e228ccf4-32f5-4f4f-a491-3e11cf73fca6.mp4

With this PR:

https://user-images.githubusercontent.com/32623900/173959949-129aff8d-e0a9-4c61-84be-6b2cc67c5b86.mp4

---

Also this PR fixes an issue with "disappearing" focus [when hero casts a spell on the adventure map. Currently in the `master` branch (notice the lost focus in the icons panel and radar which position is failed to update):

https://user-images.githubusercontent.com/32623900/174280225-cdbc5ab5-532d-4bee-9d5e-7cc60ba5f5df.mp4

With this PR:

https://user-images.githubusercontent.com/32623900/174280501-232761d3-dd8d-4f6b-ac31-46215d0dcae5.mp4

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @oleg-derevenetz , I tested your changes and found that this feature doesn't work for the case highlighted in #5533. Is it intentional?

**Comment by oleg-derevenetz** [comment]
@ihhub yes, radar highlighting now works selectively - namely on the adventure map (when user is holding the right mouse button on the hero/castle icon), in the Kingdom View window, and in the Town Portal window - that is, in places where it would be most useful.  I don't see much reason to enable this feature everywhere without exception, it's just an extra complication of the code with a real benefit in 0.01% of all use cases at best. However, if you believe that it would be useful in the use case of #5533, I can bring it back.

**Comment by ihhub** [comment]
@oleg-derevenetz , actually I am wrong: there is no such logic for other places so please do not do such changes.

**Comment by oleg-derevenetz** [comment]
@ihhub 

> actually I am wrong: there is no such logic for other places so please do not do such changes.

There is also a risk when hero's window is called from some other dialog window. Suppose that the caller window is slightly bigger than hero's window, and the bigger part partially covers some other panels (such as icon panel or status panel) in the "no interface" mode. In this case neither the `RadarUpdater` nor the hero's window `ImageRestorer` will be able to properly redraw the underlying window. Although AFAIK currently there are no such windows exist, who knows...

File: src/fheroes2/heroes/heroes_spell.cpp [hunk 1]
```diff
@@ -46,223 +46,157 @@
 #include <cassert>
 #include <memory>
 
-void DialogSpellFailed( const Spell & spell );
```
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/heroes/heroes_spell.cpp:49]: @ihhub Yes, except the changes in the `HeroesTownGate()` auxiliary function.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you so much for improving this part of the code!

**Pull request merged**